metaflow/plugins/argo/argo_client.py

```diff
@@ -152,6 +152,42 @@ def delete_workflow_template(self, name):
                     json.loads(e.body)["message"] if e.body is not None else e.reason
                 )
 
+    def terminate_workflow(self, run_id):
+        client = self._kubernetes_client.get()
+        try:
+            workflow = client.CustomObjectsApi().get_namespaced_custom_object(
+                group=self._group,
+                version=self._version,
+                namespace=self._namespace,
+                plural="workflows",
+                name=run_id,
+            )
+        except client.rest.ApiException as e:
+            raise ArgoClientException(
+                json.loads(e.body)["message"] if e.body is not None else e.reason
+            )
+
+        if workflow["status"]["finishedAt"] is not None:
+            raise ArgoClientException(
+                "Can not terminate a workflow that has already finished."
+            )
+
+        try:
+            body = {"spec": workflow["spec"]}
```

Comment:
Minor thing: should we skip the patching if we see that shutdown is already set to Terminate as we wish?  This code path probably will get exercised - when users rerun stuff (expected delays / inflight tasks etc).

Reply:
+1